js/msp.js

```diff
@@ -1485,6 +1505,14 @@ MSP.crunch = function (code) {
             buffer.push(SENSOR_CONFIG.baro_hardware);
             buffer.push(SENSOR_CONFIG.mag_hardware);
             break;
+            
+        case MSP_codes.MSP_SET_NAME:
+            for (var i = 0; i<32; i++) {
```

Comment:
This should stop at the end of the string, and not go to 32 irrespective of string length.

Also, there seems to be inconsistency with respect to maximum length for name, 100 when reading it, 32 when writing it. This should be unified.